ldap/servers/slapd/back-ldbm/db-mdb/mdb_debug.c

```diff
@@ -268,20 +272,47 @@ dbg_log(const char *file, int lineno, const char *funcname, int loglevel, char *
         if (strchr(file, '/')) {
             file = strrchr(file, '/')+1;
         }
-        append_flags(flagsstr, sizeof flagsstr, 0, "LOGLVL", loglevel, mdb_loglvl_desc);
+        append_flags(flagsstr, sizeof flagsstr, 0, "LOGLVL", loglevel & DBGMDB_LEVEL_PRINTABLE, mdb_loglvl_desc);
         slapi_log_err(SLAPI_LOG_DBGMDB, (char*)funcname, "dbmdb: %s[%d]: %s %s\n", file, lineno, flagsstr, p);
         slapi_ch_free_string(&p);
     }
 }
 
+static int
+dbg_should_log2(int loglvl, const char *filename)
+{
+    int rc = ((dbgmdb_level & loglvl) || ( filename && strcasestr(filename, BE_CHANGELOG_FILE)));
+    return rc;
+}
+
+static int
+dbg_should_log(int loglvl, MDB_dbi dbi, MDB_cursor *cursor)
+{
+    dbmdb_dbi_t *dbi1;
+
+    if (dbgmdb_level & loglvl) {
+        return 1;
+    }
+    if (dbgmdb_level & DBGMDB_LEVEL_REPL) {
+        if (cursor) {
+            dbi = mdb_cursor_dbi(cursor);
```

Comment:
No, this function is called either with a meaningful dbi or with a meaningful cursor (not both)
 so if we have a cursor then we must extract the dbi from the cursor.
    ( The reason is that dbi is not provided in the lmdb  function prototype that handle cursor _ )

Reply:
Sorry, I still miss something. If the cursor handle (dbi) is retrieved from the cursor, shouldn't do
`*dbi = mdb_cursor_dbi(cursor)` rather than
`dbi = mdb_cursor_dbi(cursor)`

Reply:
No: dbi is a MDB_dbi and not a MDB_dbi* (in fact it is a plain integer)